### PR TITLE
Limit incoming connections. 

### PR DESCRIPTION
**Pull request by arkpar**
This limits incoming connections to `max(max_peers - min_peers, min_peers / 2)`
effectively reserving at least `min_peers / 2` connections to be outbound.

Needs careful review and testing on a machine that can accept connections.

### CODE REVIEW

File: util/network-devp2p/src/host.rs [hunk 1]
```diff
@@ -566,7 +574,7 @@ impl Host {
 			self.connect_peer(&id, io);
 			started += 1;
 		}
-		debug!(target: "network", "Connecting peers: {} sessions, {} pending, {} started", self.session_count(), self.handshake_count(), started);
+		debug!(target: "network", "Connecting peers: {} sessions, {} pending, {} started", egress_count + ingress_count, self.handshake_count(), started);
```
Comment (kirushik) [inline util/network-devp2p/src/host.rs]: Why not `handshake_count`? Calling a method initiates another iteration of `session_count()`.

**Comment by kirushik** [comment]
Please don't forget to battle-test this on nightly before backporting to other release channels (which, I think, we should do eventually).

File: util/network-devp2p/src/host.rs [hunk 2]
```diff
@@ -707,12 +710,17 @@ impl Host {
 								(info.config.min_peers as usize, max_peers as usize, info.config.non_reserved_mode == NonReservedPeerMode::Deny, info.id().clone())
 							};
 
+							max_peers = max(max_peers, min_peers);
```
Comment (andresilva) [inline util/network-devp2p/src/host.rs]: Why is this needed? (assuming we validate in the config that `max_peers >= min_peers`)

**Comment by andresilva** [comment]
I'll test this tomorrow once I have a proper internet connection.

File: util/network-devp2p/src/host.rs (hunk 2 above)
Comment (arkpar) [reply util/network-devp2p/src/host.rs]: Just another sanity check, so that this code does not rely on config.

**Comment by andresilva** [comment]
This is working properly for me.

**Comment by ddorgan** [comment]
@arkpar @andresilva I installed master on a boot node and only had about 11 peers after 30 minutes. When I reverted to 1.8.4 it got 18 peers straight away.

Is there logs or debug info I can provide to you on this?

BTW the node is running with these options:
```
parity --chain kovan --log-file parity.log --auto-update=all --no-dapps --no-ui --max-peers 500 --snapshot-peers 1200 --pruning-history 1200   --jsonrpc-apis web3,eth,parity,parity_set,net,traces,rpc --no-warp --allow-ips public
```
